src/components/common/StoreReview.tsx

```diff
@@ -0,0 +1,141 @@
+            new Date(),
+            new Date(persistedLastAsked)
+          );
+          // Check if the difference is a week or more
+          const isWeekOrMoreOld =
+            differenceInDaysValue >= DAYS_TO_WAIT_BEFORE_ASKING_AGAIN;
+
+          if (!isWeekOrMoreOld) {
+            setOpen(true);
+            updateLastAsked();
+          }
+        } else {
+          updateLastAsked();
+        }
+      }
+    };
+
+    // Check if the user has already reviewed the app in the last year
+    if (
+      persistedHasReviewed &&
+      differenceInDays(new Date(), new Date(persistedHasReviewed)) >=
+        DAYS_BETWEEN_REVIEWS
+    ) {
+      setTimeout(() => {
+        getReviewDate();
+      }, 10); // Wait for 10 seconds before asking for review (To be removed once we implement it in a strategic place)
+    }
+  }, []);
+
+  const updateLastAsked = () => {
+    storage.set('@app/last-requested-review', new Date().getTime());
+  };
+
+  async function requestReview() {
+    try {
+      if (await ExpoStoreReview.hasAction()) {
+        await ExpoStoreReview.requestReview();
+        storage.set('@app/last-review-done', new Date().getTime());
+      }
+    } catch (error) {
+      console.log('> Error requesting review: ', error);
+      showToast({
+        title: t`An error occurred while requesting review`,
+        subtitle: t`Sorry about the inconvenience`,
+        type: 'error',
+      });
+    } finally {
+      setOpen(false);
+    }
+  }
+
+  function Feedback() {
+    return (
+      <FeedbackWrapper axis="y" spacing="normal" align="center">
+        <Text variant="title3">
+          <Trans>Enjoying the app?</Trans>
+        </Text>
+        <FillButton
+          variant="primary"
+          onPress={requestReview}
+          icon="heartFilled"
+          iconSide="start"
+          style={{ width: '75%' }}
+        >
+          <Trans>Loving it</Trans>
+        </FillButton>
+        <OutlineButton
+          variant="neutral"
+          onPress={() => setImprovementRequest(true)}
+          style={{ width: '75%' }}
+        >
+          <Trans>Could be better</Trans>
+        </OutlineButton>
+      </FeedbackWrapper>
+    );
+  }
+
+  return (
+    open && (
+      <BottomSheet
+        initialIndex={improvementRequest ? 1 : 0}
+        snapPoints={['25%', '35%']}
+        keyboardBehavior="fillParent"
+        onSheetChanges={(index) => {
+          // Close the bottom sheet when the user swipes down
+          if (index === -1) {
+            setOpen(false);
+          }
+        }}
+      >
+        {!improvementRequest && <Feedback />}
+        {improvementRequest && (
+          <ImprovementForm
+            setOpen={setOpen}
+            setImprovementRequest={setImprovementRequest}
+          />
+        )}
+      </BottomSheet>
+    )
+  );
+}
+
+const FeedbackWrapper = styled(Stack, {
+  width: '100%',
+  paddingHorizontal: '$normal',
+  marginTop: -20,
```

Comment:
Curious: why this negative margin?

Reply:
There is some unused space between the handle and the container, so this is moving the container up, closer to the handle

Reply:
Overcompensating a too-large margin set at the BottomSheet level 😅 Removed now